Stack/Opc.Ua.Core/Security/Certificates/CertificateFactory.cs

```diff
@@ -287,6 +304,7 @@ DateTime nextUpdate
         /// </summary>
         public static byte[] CreateSigningRequest(
             X509Certificate2 certificate,
+            // TODO: provide CertificateType to return CSR per certificate type
```

Comment:
Updated for ECC Support:

public static byte[] CreateSigningRequest(
    X509Certificate2 certificate,
    // TODO: provide CertificateType to return CSR per certificate type
    IList<String> domainNames = null
    )
{
    if (!certificate.HasPrivateKey)
    {
        throw new NotSupportedException("Need a certificate with a private key.");
    }

    CertificateRequest request = null;
    bool isECDsaSignature = X509PfxUtils.IsECDsaSignature(certificate);

    if (!isECDsaSignature)
    {
        RSA rsaPublicKey = certificate.GetRSAPublicKey();
        request = new CertificateRequest(certificate.SubjectName, rsaPublicKey,
            Oids.GetHashAlgorithmName(certificate.SignatureAlgorithm.Value), RSASignaturePadding.Pkcs1);
    }
    else
    {
        var eCDsaPublicKey = certificate.GetECDsaPublicKey();
        request = new CertificateRequest(certificate.SubjectName, eCDsaPublicKey, Oids.GetHashAlgorithmName(certificate.SignatureAlgorithm.Value));
    }
....
    if (!isECDsaSignature)
    {
        using (RSA rsa = certificate.GetRSAPrivateKey())
        {
            var x509SignatureGenerator = X509SignatureGenerator.CreateForRSA(rsa, RSASignaturePadding.Pkcs1);
            return request.CreateSigningRequest(x509SignatureGenerator);
        }
    }
    else
    {
        using (ECDsa key = certificate.GetECDsaPrivateKey())
        {
            var x509SignatureGenerator = X509SignatureGenerator.CreateForECDsa(key);
            return request.CreateSigningRequest(x509SignatureGenerator);
        }
    }
}